### ISSUE TITLE
[Sender] CastContext leaking reference to activity in 19.0.0

### ISSUE THREAD

**Issue by Ethan1983** [googlecast/CastVideos-android#99]
Hi Team,

   CastContext has a static member zzkf which in some cases ends up holding reference to activity (depending upon context type). When getSharedInstance is invoked from activity (var0), it is used to get a MediaRouter instance and this in turn is kept track in zzkf. This has changed in 19.0.0.

```
public static CastContext getSharedInstance(@NonNull Context var0) throws IllegalStateException {
        Preconditions.checkMainThread("Must be called from the main thread.");
        if (zzkf == null) {
            synchronized(zzkg) {
                if (zzkf == null) {
                    OptionsProvider var2 = zzc(var0.getApplicationContext());

                    try {
                        zzkf = new CastContext(var0, var2.getCastOptions(var0.getApplicationContext()), var2.getAdditionalSessionProviders(var0.getApplicationContext()), new zzav(MediaRouter.getInstance(var0)));
                    } catch (zzad var5) {
                        throw new RuntimeException(var5);
                    }
                }
            }
        }

        return zzkf;
    }
```

**Comment by Ethan1983** [comment]
@anjaneesh Could you have a look at this? This is not a Demo issue per se instead is on sender SDK.

**Comment by Ethan1983** [comment]
Confirmed leak via Leak Canary in the demo app too.

![Screenshot_20201117-161523](https://user-images.githubusercontent.com/1020585/99483044-6af97d00-292b-11eb-8ddf-0e65525d8d4a.png)

**Comment by Pitel** [comment]
Also happening in 21.0.1.

**Comment by Pitel** [comment]
Also happening in 21.0.1.

**Comment by Ethan1983** [comment]
Work around was just to use application context.

**Comment by strangesource** [comment]
I know this is an old issue but I stumbled upon this when debugging a memory leak and it helped me to pinpoint the problem. The issue seems to be fixed with Android Sender `21.1.0` ([ref](https://developers.google.com/cast/docs/release-notes#august-8,-2022)).

**Comment by Ethan1983** [comment]
Yeah, it's been [fixed](https://issuetracker.google.com/u/1/issues/172266576).
